test/suites/standard2/tls-configuration.robot

```diff
@@ -172,22 +177,32 @@ Check TLS Endpoints
 
     # api server, kubelet, kube controller manager and kube scheduler endpoint ports
     FOR    ${port}    IN    6443    10250    10257    10259
-        ${stdout}    ${rc}=    Openssl Connect Command    ${USHIFT_HOST}:${port}    ${TLS_AND_CIPHER_ARGS}
-        Should Be Equal As Integers    ${return_code}    ${rc}
-        IF    "${rc}" == "0"
-            Should Contain    ${stdout}    ${tls_version}, Cipher is ${cipher}
-        END
+        ${stdout}=    Wait Until Keyword Succeeds    10x    10s
+        ...    Openssl Connect Command    localhost:${port}
+        ...        ${TLS_AND_CIPHER_ARGS}
+        ...        ${return_code}
+        ...        ${tls_version}, Cipher is ${cipher}
     END
 
     # etcd endpoint, need to use cert and key because etcd requires mTLS
     Set Test Variable    ${CERT_ARG}    -cert ${APISERVER_ETCD_CLIENT_CERT}/client.crt
     Set Test Variable    ${KEY_ARG}    -key ${APISERVER_ETCD_CLIENT_CERT}/client.key
-    ${stdout}    ${rc}=    Openssl Connect Command    localhost:2379    ${TLS_AND_CIPHER_ARGS} ${CERT_ARG} ${KEY_ARG}
-    Should Be Equal As Integers    ${return_code}    ${rc}
+    Wait Until Keyword Succeeds    10x    2s
+    ...    Openssl Connect Command    localhost:2379
+    ...        ${TLS_AND_CIPHER_ARGS} ${CERT_ARG} ${KEY_ARG}
+    ...        ${return_code}
+    ...        CONNECTED
 
 Check Journal Logs
     [Documentation]    Verify system logs contain expected error messages for configuration errors
     [Arguments]    ${error_type}    ${error_case}
     ${output}    ${rc}=    Get Log Output With Pattern    ${CURSOR}    apiServer.tls
     @{expected_lines}=    Get Expected Fault Messages    ${error_type}    ${error_case}
     Compare Output Logs    ${output}    @{expected_lines}
+
+Is Cipher Available
+    [Documentation]    Check if openssl keywork is available from ssl list
+    [Arguments]    ${cipher}
+    ${rc}=    Execute Command    openssl ciphers ${cipher}
```

Comment:
You are returning both return code and stdout and storing in a single variable. This variable `rc` is now an array that holds the stdout first and the rc second. This can be seen here: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_microshift/4939/pull-ci-openshift-microshift-main-e2e-aws-tests/1924851897060036608/artifacts/e2e-aws-tests/openshift-microshift-e2e-metal-tests/artifacts/scenario-info/el94-src@standard-suite2/log.html#s1-s5
All checks are failing because of comparing the array to 0.
Adding the logic to keep going if the cipher is not available makes it silently fail every check.

Reply:
thanks, addresed in https://github.com/openshift/microshift/pull/4939/commits/ff5a733aad9282a42965be7a1d0a1c7fd49d19bc commit